src/Engine/Component/CameraComponents/CameraComponent.cpp

```diff
@@ -66,81 +73,82 @@ void CameraComponent::update(float dt) {
     rightPlanePoint = nearPlanePoint + u * nearPlaneWidth / 2.f;
     rightPlaneNormal = glm::normalize(glm::cross(v,
         rightPlanePoint - goPos));
-
-    if (isDirty) {
-        this->projection = glm::perspective(fov, Window::getAspectRatio(), near, far);
-        this->view = glm::lookAt(gameObject->getSpatial()->position(), lookAt, glm::vec3(0, 1, 0));
-        isDirty = false;
-    }
 }
 
-const bool CameraComponent::sphereInFrustum(glm::vec3 center, float radius) const {
+const bool CameraComponent::sphereInFrustum(const Sphere & sphere) const {
+    if (glm::dot(sphere.origin -    farPlanePoint,    farPlaneNormal) <= -sphere.radius) return false;
+    if (glm::dot(sphere.origin -   nearPlanePoint,   nearPlaneNormal) <= -sphere.radius) return false;
+    if (glm::dot(sphere.origin -    topPlanePoint,    topPlaneNormal) <= -sphere.radius) return false;
+    if (glm::dot(sphere.origin - bottomPlanePoint, bottomPlaneNormal) <= -sphere.radius) return false;
+    if (glm::dot(sphere.origin -   leftPlanePoint,   leftPlaneNormal) <= -sphere.radius) return false;
+    if (glm::dot(sphere.origin -  rightPlanePoint,  rightPlaneNormal) <= -sphere.radius) return false;
+
+    return true;
+
+    /*
+
     // TODO : Create loop or function to make this less lines of code
```

Comment:
Hopefully this TODO gets taken care of

Reply:
it was. left the old code commented in case Brenan wanted to look at it